ts/Core/Axis/Color/ColorAxis.ts

```diff
@@ -1001,6 +1010,21 @@ namespace ColorAxis {
 // Properties to preserve after destroy, for Axis.update (#5881, #6025).
 Array.prototype.push.apply(Axis.keepProps, ColorAxis.keepProps);
 
+addEvent(ColorAxis, 'afterGetOffset', function (): void { // (#15551)
```

Comment:
It should not be necessary with an event handler here, as we already have an override for the `getOffset` function on the ColorAxis object. It should work the same if we put this logic in the function override right after `super.getOffset()`

https://github.com/highcharts/highcharts/blob/e2238b6e0432bbbd26ca0de91de1ecd3ebedde04/ts/Core/Axis/Color/ColorAxis.ts#L500-L501